src/main/java/build/buildfarm/worker/shard/ShardWorkerInstance.java

```diff
@@ -68,6 +69,9 @@
 
 @Log
 public class ShardWorkerInstance extends AbstractServerInstance {
+  private static final Counter ioMetric =
```

Comment:
Not `static`

Reply:
Blindly copied from ShardInstance :( removed

Reply:
Counter's [register function](https://github.com/prometheus/client_java/blob/main/simpleclient/src/main/java/io/prometheus/client/SimpleCollector.java#L253) appends (registers) metrics to a static instance of the [CollectorRegistry](https://github.com/prometheus/client_java/blob/main/simpleclient/src/main/java/io/prometheus/client/CollectorRegistry.java#L28).

After I modified the ioMetric variable to be non-static, the unit tests in ShardWorkerInstanceTest began to fail. This is because each test generates a fresh ShardWorkerInstance instance, and with every instance creation, it tries to register the same metrics with the CollectorRegistry again and fails with [already register error](https://github.com/prometheus/client_java/blob/main/simpleclient/src/main/java/io/prometheus/client/CollectorRegistry.java#L55).

Although buildfarm:worker at runtime only creates single instance of `ShardWorkerInstance` and even making the ioMetric non-static will work but we have not it to be singleton. Also `ioMetric` need to be registered only once not per instance.

Reply:
@werkt Can you please merge it.

Reply:
Alright. We will need to get rid of the statics overall elsewhere.